docs/documentation/coding-style.md

```diff
@@ -0,0 +1,161 @@
+# ILIAS Coding Standard
+This coding standard is for the [ILIAS project](https://github.com/ILIAS-eLearning/ILIAS)
+
+
+**Table of Contents**
+* [PSR-2](#psr-2)
+* [Additions](#additions)
+  * [Indention](#indention)
+  * [Exceptions](#exceptions)
+  * [Abstract](#abstract)
+  * [Interfaces](#interfaces)
+  * [Operators](#operators)
+  * [Type cast](#type-cast)
+  * [Return Type Declaration](#return-type-declaration)
+  * [Strings](#strings)
+  * [Comparison](#comparison)
+  * [Comments](#comments)
+  * [Early return](#early-return)
+  * [Native Types](#native-types)
+  * [Conditions](#conditions)
+  * [Arrays](#arrays)
+
+# PSR-2 
+
+The ILIAS coding standard is aligned to the widely used and established
+[PSR-2 standard](https://www.php-fig.org/psr/psr-2/)
+of the [PHP Interop Group (PHP-FIG)](https://www.php-fig.org/).
+
+_Info: Please be aware of the [additions](#additions)
+defined for the ILIAS project that MUST be used on top of PSR-2_
+
+# Additions
+
+This document is an addition to PSR-2 Standard used in the ILIAS project.
+This section was inspired by the
+[Doctrine Coding Standards](https://github.com/doctrine/coding-standard).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Indention
+
+Divergent from the PSR-2 indentation, the code
+MUST be indent by tabs and NOT spaces.
```

Comment:
I'm personally am for spaces instead of tabs. I just added this rule, because Tabs are currently the default indentation.